src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -803,6 +803,15 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                     }
                 }
             }
+
+            configSection = (JObject)config["functionTimeout"];
+            if (configSection != null)
+            {
+                if (configSection.TryGetValue("timeout", out value))
+                {
+                    hostConfig.FunctionTimeout.Timeout = TimeSpan.Parse((string)value, CultureInfo.InvariantCulture);
```

Comment:
We should probably impose a min/max at this level. Suggestion: minimum 1 second, maximum 5 minutes. We can adjust later as needed.